### DISCUSSION TITLE
Migration from hot-reload from fast-refresh

### DISCUSSION THREAD

**Discussion by pelikhan** [gatsbyjs/gatsby#29547] [Help]
Is there a guide on how to migrate from the "hot-reload-dom" way of things to the shiny new "FAST_REFRESH" mode?

**Comment by herecydev** [comment]
It should now be enabled automatically if you're on Gatsby 2.32 onwards and are running React 17. Otherwise you can use the `GATSBY_FAST_REFRESH` env variable to set it to true

**Reply by pelikhan** [reply]
So the hot-reload stuff in the web pack configuration is not needed anymore?

**Reply by herecydev** [reply]
Which hot-reload stuff? I've never explicity had to do anything to webpack

**Reply by pelikhan** [reply]
I guess i picked up a hack from somewhere. Trying without all this.

https://github.com/gatsbyjs/gatsby/issues/11934#issuecomment-469046186

**Reply by pelikhan** [reply]
@herecydev i'm trying 3.0 and the CLI mentions

```
There are 3 other flags available that you might be interested in:
- FAST_DEV · Enable all experiments aimed at improving develop server start time
```

Is it on by default in 3.0?

**Reply by herecydev** [reply]
`FAST_DEV` is different, there are some other experiements that gatsby are running, this turns them on. I believe the specific things are `DEV_SSR`, `QUERY_ON_DEMAND` and `LAZY_IMAGES`
